lib/src/main/java/com/auth0/jwt/impl/PayloadSerializer.java

```diff
@@ -41,19 +42,65 @@ public void serialize(ClaimsHolder holder, JsonGenerator gen, SerializerProvider
                     gen.writeEndArray();
                 }
             } else {
-                gen.writeFieldName(e.getKey());
-                if (e.getValue() instanceof Date) { // true for EXPIRES_AT, ISSUED_AT, NOT_BEFORE
-                    gen.writeNumber(dateToSeconds((Date) e.getValue()));
-                } else {
-                    gen.writeObject(e.getValue());
-                }
+                handleSerialization(e, gen);
             }
         }
 
         gen.writeEndObject();
     }
 
-    private long dateToSeconds(Date date) {
-        return date.getTime() / 1000;
+    /**
+     * Serializes {@linkplain Instant} to epoch second values, traversing maps and lists as needed.
+     * @param entry the entry to serialize
+     * @param gen the JsonGenerator to use for JSON serialization
+     * @throws IOException
+     */
+    private void handleSerialization(Map.Entry<String, Object> entry, JsonGenerator gen) throws IOException {
+        gen.writeFieldName(entry.getKey());
+        if (entry.getValue() instanceof Instant) { // EXPIRES_AT, ISSUED_AT, NOT_BEFORE, custom Instant claims
```

Comment:
All these _if and elses_ can be replaced with a call to `serialize(entry.getValue(), gen)`. I'd even consider if 2 methods need to exists (handleSerialization and serialize)